code/modules/mob/living/carbon/human/human_damage.dm

```diff
@@ -10,9 +10,17 @@
 
 	health = maxHealth - getBrainLoss()
 
-	//TODO: fix husking
-	if(((maxHealth - getFireLoss()) < config.health_threshold_dead) && stat == DEAD)
-		ChangeToHusk()
+	if(stat == DEAD)
+		if(getFireLoss() > maxHealth * 1.5)
+			ChangeToHusk()
+		if(getFireLoss() > maxHealth * 2.5)
+			ChangeToSkeleton()
+		if(getFireLoss() > maxHealth * 3)
+			for(var/obj/item/I in src)
+				drop_from_inventory(I)
+			visible_message(SPAN_WARNING("\The [src] burns away into a pile of ash."))
+			new /obj/effect/decal/cleanable/ash(get_turf(src))
```

Comment:
I recommend just calling dust()) here. It will leave species-based remains and will delete the mob as well, so you can remove the qdel in the end of this proc.